src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -95,16 +102,25 @@ public TextSubmission handleTextSubmission(TextSubmission textSubmission, TextEx
      *
      * @param textSubmission the submission that should be saved
      * @param participation  the participation the participation the submission belongs to
+     * @param principal      the principal of the user
      * @return the textSubmission entity that was saved to the database
      */
     @Transactional(rollbackFor = Exception.class)
-    public TextSubmission save(TextSubmission textSubmission, StudentParticipation participation) {
+    public TextSubmission save(TextSubmission textSubmission, StudentParticipation participation, Principal principal) {
         // update submission properties
         textSubmission.setSubmissionDate(ZonedDateTime.now());
         textSubmission.setType(SubmissionType.MANUAL);
         textSubmission.setParticipation(participation);
         textSubmission = textSubmissionRepository.save(textSubmission);
 
+        // versioning of submission
+        try {
+            submissionVersionService.save(textSubmission, principal.getName());
```

Comment:
please only invoke this for team exercises (pass TeamExercise in the method)

Reply:
Done